### DISCUSSION TITLE
Next js 13 onRouteChange events

### DISCUSSION THREAD

**Discussion by 1mehdifaraji** [vercel/next.js#43950] [Help] [answered]
How and where can I use router events with Next js 13 ?

```
useEffect(() => {

    const handleRouteStart = () =>
      NProgress.configure({ showSpinner: true }).start();
    const handleRouteDone = () =>
      NProgress.configure({ showSpinner: true }).done();
 
    Router.events.on("routeChangeStart", handleRouteStart);
    Router.events.on("routeChangeComplete", handleRouteDone);

    return () => {

      Router.events.off("routeChangeStart", handleRouteStart);
      Router.events.off("routeChangeComplete", handleRouteDone);

    };
  }, []);
```

**Comment by icyJoseph** [comment, marked as answer]
The architecture change made router events slightly more difficult to implement

![Screenshot 2022-12-12 at 07 14 32](https://user-images.githubusercontent.com/21013447/206973205-0c168ef9-6067-4b0c-868b-248a95ac5993.png)

You can make an ad-hoc solution by using `usePathname` inside a client component, and placing that as high on the tree as you can, but that'd need some `use-effect` trickery. More on these discussion comments: https://github.com/vercel/next.js/discussions/41745#discussioncomment-3986452
